server/auth/sso/sso.go

```diff
@@ -185,21 +187,32 @@ func newSso(
 	if c.IssuerAlias != "" {
 		lf["issuerAlias"] = c.IssuerAlias
 	}
+	var filterGroupsRegex []*regexp.Regexp
+	if c.FilterGroupsRegex != nil && len(c.FilterGroupsRegex) > 0 {
+		for _, regex := range c.FilterGroupsRegex {
+			compiledRegex, err := regexp.Compile(regex)
+			if err != nil {
+				return nil, fmt.Errorf("failed to compile sso.filterGroupRegex: %s %w", regex, err)
+			}
+			filterGroupsRegex = append(filterGroupsRegex, compiledRegex)
+		}
+	}
 	log.WithFields(lf).Info("SSO configuration")
```

Comment:
`filterGroupsRegex` should also be added to `lf` above.

or perhaps better, the whole code block should be moved above the `lf` assignment as those statements were intended to be together. this got spliced in between the `lf` code block

can put newlines around this code block and that code block so that it is more clear that they are blocks of related code

Reply:
done